test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,194 @@
+                     NetworkFilterNames::get().DirectResponse,
+                     NetworkFilterNames::get().DubboProxy,
+                     NetworkFilterNames::get().SniCluster};
+  }
+  return filter_names_;
+}
+
+void UberFilterFuzzer::reset() {
+  // Reset some changes made by current filter on some mock objects.
+
+  // Close the connection to make sure the filter's callback is set to nullptr.
+  read_filter_callbacks_->connection_.raiseEvent(Network::ConnectionEvent::LocalClose);
+  // Clear the filter's raw pointer stored inside the connection_ and reset the connection_.
+  read_filter_callbacks_->connection_.callbacks_.clear();
+  read_filter_callbacks_->connection_.bytes_sent_callbacks_.clear();
+  read_filter_callbacks_->connection_.state_ = Network::Connection::State::Open;
+}
+void UberFilterFuzzer::perFilterSetup(const std::string& filter_name) {
+  // Set up response for ext_authz filter
+  if (filter_name == NetworkFilterNames::get().ExtAuthorization) {
+
+    async_client_factory_ = std::make_unique<Grpc::MockAsyncClientFactory>();
+    async_client_ = std::make_unique<Grpc::MockAsyncClient>();
+
+    ON_CALL(*async_client_, sendRaw(_, _, _, _, _, _))
+        .WillByDefault(testing::WithArgs<3>(Invoke([&](Grpc::RawAsyncRequestCallbacks& callbacks) {
+          Filters::Common::ExtAuthz::GrpcClientImpl* grpc_client_impl =
+              dynamic_cast<Filters::Common::ExtAuthz::GrpcClientImpl*>(&callbacks);
+          const std::string empty_body{};
+          const auto expected_headers =
+              Filters::Common::ExtAuthz::TestCommon::makeHeaderValueOption({});
+          auto check_response = Filters::Common::ExtAuthz::TestCommon::makeCheckResponse(
+              Grpc::Status::WellKnownGrpcStatus::Ok, envoy::type::v3::OK, empty_body,
+              expected_headers);
+          // Give response to the grpc_client by calling onSuccess().
+          grpc_client_impl->onSuccess(std::move(check_response), span_);
+          return async_request_.get();
+        })));
+
+    EXPECT_CALL(*async_client_factory_, create()).WillOnce(Invoke([&] {
+      return std::move(async_client_);
+    }));
+
+    EXPECT_CALL(factory_context_.cluster_manager_.async_client_manager_,
+                factoryForGrpcService(_, _, _))
+        .WillOnce(Invoke([&](const envoy::config::core::v3::GrpcService&, Stats::Scope&, bool) {
+          return std::move(async_client_factory_);
+        }));
+  }
+}
+void UberFilterFuzzer::fuzzerSetup() {
+  // Setup process when this fuzzer object is constructed.
+  // For a static fuzzer, this will only be executed once.
+
+  // Get the pointer of read_filter when the read_filter is being added to connection_.
+  read_filter_callbacks_ = std::make_shared<NiceMock<Network::MockReadFilterCallbacks>>();
+  ON_CALL(read_filter_callbacks_->connection_, addReadFilter(_))
+      .WillByDefault(Invoke([&](Network::ReadFilterSharedPtr read_filter) -> void {
+        read_filter_ = read_filter;
```

Comment:
How come this member never needs to be reset at the end of a run?

Reply:
Yes this needs to be reset. Fixed!